### PR TITLE
Project Happy Thoughts API - Mathias T - Review wanted

### PR DESCRIPTION
**Pull request by Nesiht**
Fun to connect it to previous project and really cool see the results when everything is working correctly. Really good project that helps to see the connection between frontend with backend.

[Backend](https://plunta-happy-thought-api.herokuapp.com/)
[Frontend](https://relaxed-shockley-ec05e0.netlify.app/)

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -2,28 +2,62 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import Thought from './model/Thought'
```
Comment (KarolCislowski) [inline server.js:5]: External Model 🥇 👍

File: server.js [hunk 2]
```diff
@@ -2,28 +2,62 @@ import express from 'express'
...
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec()
+  res.json(thoughts)
+})
```
Comment (KarolCislowski) [inline server.js:36]: according to documentation `get "/"` need to return 20 thoughts,

File: server.js [hunk 3]
```diff
@@ -2,28 +2,62 @@ import express from 'express'
...
+  const { message } = req.body
+  const thought = new Thought({ message })
+
+  try{
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought)
+  } catch (err) {
+    res.status(400).json({ message: 'Could not save thought to the databse' , error:err.errors})
+  }
+})
```
Comment (KarolCislowski) [inline server.js:48]: according to documentation `post "/"` need to create a new one.

File: server.js (hunk 2 above)
Comment (Nesiht) [reply server.js]: Updated.

File: server.js (hunk 3 above)
Comment (Nesiht) [reply server.js]: Updated.

**Pull request closed without merging**